src/valid/expression.rs

```diff
@@ -755,6 +755,9 @@ impl super::Validator {
                 if !condition_good || accept_inner != reject_inner {
                     return Err(ExpressionError::InvalidSelectTypes);
                 }
+                if let Ti::Pointer { .. } = *accept_inner {
```

Comment:
I think this is a wrong level to check this. We are checking for pointers, but what if it's a texture, for example?
Instead, we need to check the derived type flags on this and ensure this is `DATA | SIZED`